### PR TITLE
Add CVE scan workflow

### LINKED ISSUE #4521: Add CVE scan workflow for PR
**Describe the feature request**
As a developer/maintainer of Tanzu Framework repository, I would like to have CVE scan as part of PR checks, so that I know if my change introduces any vulnerabilities.

**Describe alternative(s) you've considered**
NA

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- (X) Security
- ( ) Test and Release
- ( ) User Experience
- (X) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by avi-08**
### What this PR does / why we need it

Adding a CVE scan workflow:

- uses trivy github action for vulnerability scan on PR
- [CVSSv3 scores >= 7.0](https://www.first.org/cvss/v3.0/specification-document#Qualitative-Severity-Rating-Scale) are considered for severity reporting
- uploads the report to Github security tab > Code Scanning.
- Scan reports can be filtered based on PR#

Bumped `gopkg.in/yaml.v3` version to fix CVE captured in an [earlier run](https://github.com/vmware-tanzu/tanzu-framework/actions/runs/4552196661) of this workflow

### Which issue(s) this PR fixes

Fixes #4521 

### Describe testing done for PR

The "CVE Scan" action is triggered and report is added to Github [security tab](https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning?query=pr%3A4530+is%3Aopen).

### Release note

```release-note

```

### CODE REVIEW

File: hack/cve/trivy.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Ref: https://aquasecurity.github.io/trivy/v0.33/docs/references/customization/config-file
```
Comment (rajathagasthya) [inline hack/cve/trivy.yaml:1]: nit: I think we can keep config files like this at the top level directory (see others like `.codecov.yaml` or `.golangci.yaml`). Just make sure to prefix with a `.`.

File: hack/cve/trivy.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+format: 'sarif'
+output: 'trivy-results.sarif'
+severity:
+  - CRITICAL
+  - HIGH
+# Scan Options
+scan:
+  skip-dirs:
+    - docs
+    - hack
```
Comment (rajathagasthya) [inline hack/cve/trivy.yaml:15]: `hack` also has code, so maybe it's useful there as well?

File: .github/workflows/cve_scan.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: CVE Scan
```
Comment (rajathagasthya) [inline .github/workflows/cve_scan.yaml:1]: I'm wondering if there is overlap with CodeQL scanning already setup for this repository.

File: hack/cve/trivy.yaml (hunk 1 above)
Comment (avi-08) [reply hack/cve/trivy.yaml:1]: I'll move the config to project root.

File: hack/cve/trivy.yaml (hunk 2 above)
Comment (avi-08) [reply hack/cve/trivy.yaml:15]: Makes sense. Will remove this.

**Review by joshuatcasey — APPROVED** [review]
I'm approving on behalf of @vmware-tanzu/tkg-iam-owners , the only IAM file change appears to be in `pinniped-components/hack/tools/go.mod` (and the ` go.sum`), and the change is fine.

File: .github/workflows/cve_scan.yaml (hunk 3 above)
Comment (vuil) [reply .github/workflows/cve_scan.yaml:1]: @avi-08 where in the security tab will there results show up?

Comment (avi-08) [reply .github/workflows/cve_scan.yaml:1]: @vuil The reports can be accessed from Security > Code Scanning > PR# filter. For this PR: https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning?query=pr%3A4530+is%3Aopen, it shows 0 open and 18 closed issues

File: hack/cve/trivy.yaml (hunk 2 above)
Comment (avi-08) [reply hack/cve/trivy.yaml]: Updated in latest changes.

File: .github/workflows/cve_scan.yaml (hunk 3 above)
Comment (rajathagasthya) [reply .github/workflows/cve_scan.yaml:1]: non-blocking: If we can add an automated comment on this PR with a link mentioned above so we can view the alerts easily, that would be great.

Comment (avi-08) [reply .github/workflows/cve_scan.yaml:1]: Added the comment step in workflow

**Review by yharish991 — APPROVED** [review]
LGTM!!

**Pull request merged**